components/collator/src/elements.rs

```diff
@@ -623,8 +670,13 @@ impl Default for NonPrimary {
 /// when `upcoming[0]` is an undecomposed starter, we don't
 /// need the ccc yet, and when lookahead has already done the
 /// trie lookups, we don't need `trie_value`, as it is implied
-/// by ccc. Optimizing size by this observation seems
-/// unnecessary, since `upcoming` is typically very short.
+/// by ccc.
+///
+/// TODO: This struct carries redundant information, and
```

Comment:
nit (optional): create issue and update as `TODO(#issue)`

Reply:
Done